### ISSUE TITLE
Double-Sided materials intermittently write to depth buffer

### ISSUE THREAD

**Issue by BStringhamVRSK** [google/filament#1935]
**Description:**
The back faces of an object using a material that is set as double-sided will intermittently write to the depth buffer.  The material's "**doubleSided**" property is explicitly set to _false_ in the material file, then set to _true_ at run-time.

**To Reproduce**
The scene consists of multiple objects, all using the same material, but **_different instances_** of it.  Some of the instances have doubleSided set to _true_, some are set to _false_.
When viewing the scene from various viewpoints, some back faces that should be writing to the z-buffer will not, resulting in geometry showing through that should not.

**Expected behavior**
Back faces using a material with "doubleSided" set to true should always write to the z-buffer.  The first image, below, shows a roof exhibiting the problem.  The second image is mostly correct, with the back faces of the main roof rendering into the depth buffer.

**Screenshots**
![Z-Buffer_Fail](https://user-images.githubusercontent.com/40642629/70186080-57b4b980-16a8-11ea-8f0a-69cce2c8d50f.JPG)
![Z-Buffer_OK](https://user-images.githubusercontent.com/40642629/70186081-57b4b980-16a8-11ea-85b9-bf8dd54000c8.JPG)

**Desktop (please complete the following information):**
 - OS: Windows 10
 - GPU: AMD FirePro D500 (FireGL V)
 - Backend: OpenGL

**Additional context**
I verified the issue using RenderDoc to verify that the back-faces were not always rendering to the depth buffer.  The symptoms lead me to believe that the **doubleSided** attribute is not always being respected during the depth buffer pass, possibly because some of the objects in the scene use instances of that material with **doubleSided** enabled, some with it disabled.

Here is the material definition file:
```
material {
    name : Lit_Textured,
    requires : [
        uv0
    ],
    shadingModel : lit,
	doubleSided : false,
	colorWrite : true,
	depthWrite : true,
	depthCulling : true,
	culling : back,
    parameters : [
        {
			type : float4,
			name : baseColor
        },
        {
            type : sampler2d,
            name : albedo
        },
        {
            type : sampler2d,
            name : roughness
        },
        {
            type : sampler2d,
            name : metallic
        },
        {
            type : sampler2d,
            name : normal
        },
        {
            type : sampler2d,
            name : ao
        }
    ],
}

fragment {
    void material(inout MaterialInputs material) {
		float4 rawNormal = texture(materialParams_normal, getUV0()).xyzw * 2.0 - 1.0;
		float3 normal;
		normal.x = rawNormal.x * rawNormal.w;
		normal.y = rawNormal.y;
		float2 normXYSqr = normal.xy * normal.xy;
		normal.z = sqrt(1.0 - normXYSqr.x - normXYSqr.y);
        material.normal = normal;
        prepareMaterial(material);
		float4 texColor = texture(materialParams_albedo, getUV0());
		texColor.xyz = texColor.xyz * texColor.xyz;
        material.baseColor = materialParams.baseColor * texColor;
        material.roughness = texture(materialParams_roughness, getUV0()).r;
        material.metallic = texture(materialParams_metallic, getUV0()).r;
        material.ambientOcclusion = texture(materialParams_ao, getUV0()).r;
    }
}

```

**Comment by prideout** [comment]
This might be fixed by https://github.com/google/filament/pull/1936.  We were not correctly honoring the culling state associated with the material instance.

**Comment by prideout** [comment]
@BStringhamVRSK would you be willing to try the above fix?

**Comment by BStringhamVRSK** [comment]
Sure!  It might take me a little bit to work around our build process to inject my own libraries, but I will let you know what I discover.

**Comment by BStringhamVRSK** [comment]
That appears to have worked.  Glad to confirm the fix.

**Comment by romainguy** [comment]
Thanks for checking!

**Comment by prideout** [comment]
Fixed by #1936
